Parsers/ASimWebSession/Parsers/ASimWebSessionCiscoMeraki.yaml

```diff
@@ -0,0 +1,228 @@
+Parser:
+  Title: Web Session ASIM parser for Cisco Meraki
+  Version: '0.1'
+  LastUpdated: Jun 20 2023
+Product:
+  Name: Cisco Meraki
+Normalization:
+  Schema: WebSession
+  Version: '0.2.6'
+References:
+- Title: ASIM Web Session Schema
+  Link: https://aka.ms/ASimWebSessionDoc
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
+- Title: Cisco Meraki Documentation
+  Link: https://documentation.meraki.com/
+Description: |
+  This ASIM parser supports normalizing Cisco Meraki logs to the ASIM Web Session normalized schema.
+ParserName: ASimWebSessionCiscoMeraki
+EquivalentBuiltInParser: _ASim_WebSession_CiscoMeraki
+ParserParams:
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let ActionLookup = datatable (action: string, DvcAction: string)  [
+    'allow', 'Allow',
+    'log', 'Allow',
+    'accept', 'Allow',
+    'block', 'Deny',
+    'deny', 'Deny',
+    'quarantine', 'Deny'
+  ];
+  let parser=(disabled: bool=false) {
+      union isfuzzy=true
+          (
+          meraki_CL
+          | project-rename LogMessage =  Message
+          ),
+          (
+          Syslog
+          | project-rename LogMessage =  SyslogMessage
+          )
+      | where not(disabled) and (LogMessage has "urls" or LogMessage has_all("security_event", "security_filtering_file_scanned"))
+      | extend Parser = extract_all(@"(\d+.\d+)\s([\w\-\_]+)\s([\w\-\_]+)\s([\S\s]+)$", dynamic([1, 2, 3, 4]), LogMessage)[0]
```

Comment:
can we use inbuilt function instead of regex to extract this value?

Reply:
@jayeshprajapaticrest I couldn't find this change. can you please check and comment

Reply:
Event format1:
2019-07-16 04:00:30 [Local0.Info](http://local0.info/) 172.40.20.177 1 1561036264.565291108 australia_sydney security_event security_filtering_file_scanned url=http://tlu123.dl123.delivery.mp.microsoft.com/filestreamingservice/files/36ec4eb7-46dd-4aeb-990e-b6d32f7ed567?P1=1561036860& P2=402&P3=2&P4=Zj3qRDR5CbzfWlP8BuYg%2bUlTon0XE774ExEEquiawstLAJ2%2bQm3OoWLcwz3HBt8qp3r3buVRVoT5BQcUCcNlXw%3d%3d src=[172.16.10.98:64160](http://172.16.10.98:64160/) dst=[200.188.210.42:180](http://200.188.210.42:180/) mac=20:1C:BC:B2:0F:20 name='' sha256093e4fc218b27e58e2fede7b8cb044d48d66995ae785bbc186a9df5ae08ca4f7 disposition=malicious action=block

Event format2:
<134>1 1685956046.281934865 CDSAPF3_11 events type=association radio='1' vap='3' client_mac='F2:ED:56:6A:98:6A' last_known_client_ip='172.20.12.194' band='5' channel='104' rssi='38' identity='zuber.chataiwala' aid='1305860724'

We double checked and we have the different event formats as above so we are not able to do the parsing using inbuilt function like parse to get the required fields.

Reply:
@jayeshprajapaticrest can you test with this?

| parse LogMessage with * " " DatetimeEpoch1:long "." DatetimeEpoch2:long " " Device:string " " LogType:string " " Substring:string

Also, I noticed that 'australia_sydney' is coming as Device name, is that postion correct?